docs/plans/2026-05-07-dlt-3281-eslint-deprecated-class-props.md

```
@@ -0,0 +1,447 @@
+**Key Decisions / Notes:**
+
+- **Visitor:** use `parserServices.defineTemplateBodyVisitor({ VElement(node) { ... } })`. We need access to the start tag's full attribute list to support cross-attribute merge logic in Task 3 — `VElement` gives us `node.startTag.attributes`. Pattern matches `lib/rules/deprecated-stack-alignment-classes.js:50-108`.
+- **Tag detection:** match `node.rawName` or `node.name` against `^dt-` (kebab) or `^Dt[A-Z]` (PascalCase). Skip non-Dialtone tags entirely.
+- **Tag-name normalization:** kebab-to-PascalCase function. Example: `dt-feed-item-pill` → `DtFeedItemPill`. Reuse Vue's standard conversion: split on `-`, capitalize each segment, join.
+- **Attribute matching:** for each attribute in `node.startTag.attributes`, check if the attribute name (or directive argument for `:`-bound) matches one of the 6 deprecated names (3 props × 2 casings). Use a const set. Example for static: `attr.directive === false && attr.key.name === 'root-class'`. For dynamic: `attr.directive === true && attr.key.argument?.name === 'root-class'`.
+- **Module-level data load (defensive):** at the top of the rule file, attempt `require('@dialpad/dialtone-vue/component-documentation.json')` inside try/catch. On failure, set `components = []` and emit `console.warn('[eslint-plugin-dialtone] Could not load component-documentation.json from @dialpad/dialtone-vue. The deprecated-class-props rule will be skipped.')` once. The `require` call is what tests will stub via `proxyquire` — the rule must use this exact import path (no indirection) so the stub matches.
+- **Lookup helper:** `componentDeclaresProp(displayName, propName)` — `components.find(c => c.displayName === displayName)?.props?.some(p => p.name === propName) ?? false`. Cache the lookup if perf shows up as an issue (premature for now — N=~60 components, list traversal is fine).
+- **Reporting:** report on the offending `VAttribute` node. Message: `"DtFoo does not accept a 'rootClass' prop. Use the native 'class' attribute instead. See: https://dialtone.dialpad.com/guides/migration/component-props/"`.
+- **No fix in this task** — `meta.fixable: null` for now. Task 3 flips to `'code'` and adds the fixer.
+- **Test setup:** mirror `tests/lib/rules/deprecated-stack-alignment-classes.js` AND inject mocked component data via `proxyquire`:
+  ```js
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should be surrounded by blank lines

(MD031, blanks-around-fences)